backends/tc/backend.cpp

```diff
@@ -271,6 +271,53 @@ void ConvertToBackendIR::postorder(const IR::P4Action *action) {
     }
 }
 
+void ConvertToBackendIR::updateConstEntries(const IR::P4Table *t, IR::TCTable *tabledef) {
+    // Check if there are const entries.
+    auto entriesList = t->getEntries();
+    if (entriesList == nullptr) return;
+    auto keys = t->getKey();
+    for (auto e : entriesList->entries) {
+        if (keys == nullptr) {
+            return;
+        }
+        auto keyset = e->getKeys();
+        if (keyset->components.size() != keys->keyElements.size()) {
+            ::error("No of keys in const_entries should be same as no of keys in the table.");
+            return;
+        }
+        ordered_map<cstring, cstring> keyList;
+        for (size_t itr = 0; itr < keyset->components.size(); itr++) {
+            auto keyElement = keys->keyElements.at(itr);
+            auto keyString = keyElement->expression->toString();
+            auto annotations = keyElement->getAnnotations();
+            if (annotations) {
+                if (auto anno = annotations->getSingle("name")) {
+                    keyString = anno->expr.at(0)->to<IR::StringLiteral>()->value;
+                }
+            }
+            auto keySetElement = keyset->components.at(itr);
+            auto key = keySetElement->toString();
+            if (keySetElement->is<IR::DefaultExpression>()) {
+                key = "default";
+            } else if (keySetElement->is<IR::Constant>()) {
+                big_int kValue = keySetElement->to<IR::Constant>()->value;
+                int kBase = keySetElement->to<IR::Constant>()->base;
+                std::stringstream value;
+                std::deque<char> buf;
+                do {
+                    const int digit = static_cast<int>(static_cast<big_int>(kValue % kBase));
+                    kValue = kValue / kBase;
+                    buf.push_front(Util::DigitToChar(digit));
+                } while (kValue > 0);
+                for (auto ch : buf) value << ch;
+                key = value.str().c_str();
+            }
+            keyList.emplace(keyString, key);
+        }
```

Comment:
no other expressions supported?
range, mask etc

Reply:
addressed

Reply:
Hope you checked all possible values